feat: docs for local dev and deployment

alpuga opened the pull request:
<!-- ELLIPSIS_HIDDEN -->


> [!IMPORTANT]
> Adds documentation for deployment and local development setup, and updates GitHub workflow for production deployment.
> 
>   - **Documentation**:
>     - Adds `docs/deployment.md` detailing the deployment process from development to production, including prerequisites, staging, and production deployment steps.
>     - Adds `docs/local_dev_setup.md` providing a guide for setting up local development, including environment configuration and running the application.
>   - **GitHub Workflow**:
>     - Updates `.github/workflows/update-runtime-ecr.yaml` to include production in the build and push process, changing the name and run-name to "Build and push parameterized runtime to staging and prod".
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=abstractoperators%2Faiden&utm_source=github&utm_medium=referral)<sup> for 51b196d8c82bf279079d35141583c0e144c79a35. You can [customize](https://app.ellipsis.dev/abstractoperators/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>


<!-- ELLIPSIS_HIDDEN -->

kgang commented on docs/deployment.md:
```diff
@@ -0,0 +1,150 @@
+
+Repeat steps 6-9 for the production environment:
+
+1. Execute "build and push parameterized runtime to staging" action
```
can you change the name of that github workflow to say that it's updating both? tthanks

kgang reviewed (approved):
LGTM!

alpuga commented:
I updated the name of the github workflow in update-runtime-ecr.yaml to include staging and prod. Then I updated the deployment doc to account for this name change. 

Should clarify the confusion that we use the same workflow to build our runtimes for both staging and prod.

crypto-cooker reviewed (approved):
LGTM
